tianshou/data/batch.py

```diff
@@ -52,17 +52,19 @@ def _is_number(value: Any) -> bool:
 
 def _to_array_with_correct_type(v: Any) -> np.ndarray:
     # convert the value to np.ndarray
-    # convert to np.object data type if neither bool nor number
     v = np.asanyarray(v)
```

Comment:
``np.asanyarray`` fails in the presence of cuda tensor ``torch.ones(1, 2, device='cuda')``. Maybe this pr can fix it by the way?

Reply:
Good point. But I checked the code and it should raise an appropriate exception in any case.

Reply:
But it should not raise an exception.

Reply:
> But it should not raise an exception.

To be clearer: It should raise an appropriate exception **when appropriate**. :smile:

Reply:
> But it should not raise an exception.

Agree. ``_to_array_with_correct_type `` should deal with cuda tensors correctly, rather than raising an exception.

Reply:
> Agree. _to_array_with_correct_type should deal with cuda tensors correctly, rather than raising an exception.

It is just wrong. `_to_array_with_correct_type` is never dealing with torch.Tensor objects in a relevant scenario. So it makes no sense to do anything else at `_to_array_with_correct_type` level. It will be handle by `_parse_value` later anyway.

Reply:
> _to_array_with_correct_type is never dealing with torch.Tensor objects in a relevant scenario (except for raising exceptions).

Ok, that's reasonable. tensors are handled by ``_parse_value``.